### PR TITLE
Add or replace license headers in all relevant files

### PR DESCRIPTION
**Pull request by alexgleith**
### Reason for this pull request

For the Open Data Cube to become an OSGeo Project, it needs to include license headers in source code files.

### Proposed changes

- Add license header text to relevant files

### CODE REVIEW

**Comment by Kirill888** [comment]
@alexgleith can you please squash all those commits, so we only have one commit to ignore in the future.

**Comment by omad** [comment]
Can we make it a bit tighter, 8 lines seems excessive. Is it legitimate to leave dates out entirely? At the least, we should only specify a starting date, and put the current year in new files, which seems to be what other projects do.

Something like:

```
# This file is part of the Open Data Cube. See https://opendatacube.org for more information.
#
# (c) 2019 ODC Contributors
# SPDX-License-Identifier: Apache-2.0
```

**Comment by omad** [comment]
As examples, Geoserver uses:
```
/* (c) 2017 Open Source Geospatial Foundation - all rights reserved
 * This code is licensed under the GPL 2.0 license, available at the root
 * application directory.
 */
```
and QGIS has this behemoth:
```
"""
***************************************************************************
    QtPrintSupport.py
    ---------------------
    Date                 : November 2015
    Copyright            : (C) 2015 by Matthias Kuhn
    Email                : matthias at opengis dot ch
***************************************************************************
*                                                                         *
*   This program is free software; you can redistribute it and/or modify  *
*   it under the terms of the GNU General Public License as published by  *
*   the Free Software Foundation; either version 2 of the License, or     *
*   (at your option) any later version.                                   *
*                                                                         *
***************************************************************************
"""
```

**Review by whatnick — APPROVED** [review]
Looks fairly compact. Approved 👍

**Comment by alexgleith** [comment]
New headers for Python files are:
```
# This file is part of the Open Data Cube, see https://opendatacube.org for more information
#
# Copyright (c) 2015-2020 ODC Contributors
# SPDX-License-Identifier: Apache-2.0
```

Similar for YAML

File: integration_tests/data/dataset_add/datasets.yml [hunk 1]
```diff
@@ -1,11 +1,9 @@
-# Generated with `yaml.safe_dump_all(gen_dataset_test_dag(ii, force_tree=True) for ii in range(1,3))`
```
Comment (Kirill888) [inline integration_tests/data/dataset_add/datasets.yml]: these bits are useful comments

File: datacube/model/schema/ingestor-config-type-schema.yaml [hunk 2]
```diff
@@ -1,4 +1,9 @@
-"$schema": "http://json-schema.org/draft-04/schema#"
```
Comment (Kirill888) [inline datacube/model/schema/ingestor-config-type-schema.yaml]: this is probably essential but got removed

Comment (alexgleith) [reply datacube/model/schema/ingestor-config-type-schema.yaml]: Weird.

**Comment by alexgleith** [comment]
Changed to only run over `.py` files.

File: docs/_static/custom.css [hunk 3]
```diff
@@ -1,3 +1,14 @@
...
+This file is part of the Open Data Cube, see https://opendatacube.org for more information
+
+Copyright (c) 2015-2020 ODC Contributors
+SPDX-License-Identifier: Apache-2.0
+*/
+/*
+This file is part of the Open Data Cube, see https://opendatacube.org for more information
+
+Copyright (c) 2015-2020 ODC Contributors
+SPDX-License-Identifier: Apache-2.0*/
```
Comment (omad) [inline docs/_static/custom.css]: Double license in this file...

**Review by omad — APPROVED** [review]
Looks good, thanks!

File: docs/_static/custom.css (hunk 3 above)
Comment (alexgleith) [reply docs/_static/custom.css]: Oh, css...

**Pull request merged**